eahub/templates/components/feedback.html

```diff
@@ -0,0 +1,79 @@
+{% block feedback %}
+{% load static %}
+  
+  <div id="overlay"></div>
+
+  <div id="feedback">
+  <!-- Button trigger modal -->
+  <button type="button" class="btn feedbackbtn" onClick="togglemodal()">
+      Feedback?
+    </button>
+    
+    <!-- Modal -->
+    <div class="modal fade" id="feedbackModal" tabindex="-1">
+      <div class="modal-content">
+          <div class="modal-header">
+            <h5 class="modal-title">Leave Feedback</h5>
+          </div>
+          <div class="modal-body">
+
+            <form id="form" class="form-group" method="post" action="/feedback/submit/">
+              <p>Do you have any feedback for us?</p>
+              {% csrf_token %}
+
+              <textarea class="form-control" name="message" cols="37" rows="5" required="required"></textarea>
+              <div id="errormsg" style="display:none" class='alert alert-danger'>
+                Feedback message field is required</div>
+            
+              <label for="email">Email (optional)</label>
+              <input class="form-control" type="text" name="email">
+              <input id="page_url" type="hidden" name="page_url" value="">
+              
+              <div class="modal-footer">
+                <button  id="closebtn" type="button" class="btn btn-secondary" onClick="togglemodal()">Close</button>
+                <input onClick="ajaxsubmit()" value="Submit" class="btn submitbtn" />
+              </div>
+              </form>
+
+              <div id="successmsg" style="display:none">
+                Your feedback has been received, thank you!</div>
+          </div>     
+        </div>
+    </div>  
+
+  </div>
+
+  
+  <script>
+  function togglemodal(){
+    var m = document.getElementById("feedbackModal")
+    m.classList.toggle("show");
+    var element = document.getElementById("feedback");
+    element.classList.toggle("active");
+    document.getElementById("overlay").classList.toggle("active");    
+  }
+  document.getElementById('page_url').value = document.referrer;
+
+  async function ajaxsubmit(){
+    var form = document.getElementById('form');
+    const response = await fetch(form.action, {
+      method: form.method,
+      body: new URLSearchParams(new FormData(form)),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).then(response => {
+      if(response.status == "200") {
+        document.getElementById("successmsg").style.display = "block";
+        document.getElementById("form").style.display = "none";
+        console.log("success");
+      }
+      if(response.status == "418") {
```

Comment:
oh btw in JS it's a requirement to use `===` instead of `==` unless there's a critical reason for the latter - in short `==` is alive mainly for the sake of old browsers as <ie11, since it doesn't really work correctly

Reply:
I changed them to ===, but now the submit button has stopped working. Would I need to change the type of "418" from a string to something else if I use ===?

Reply:
quite probably - the comparison coercion there isn't consistent and it's been a bane of javascript for a while, below you can find few examples of how it works in practices:
```javascript
true == '1' // true
true == 'true' // false
0 == '' // true
undefined == null // true
```

Reply:
I guess `response.status` is a `Number`?